deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -1109,9 +1110,9 @@ phase_stop_member(StreamId, #{node := Node,
     end.
 
 phase_start_writer(StreamId, #{epoch := Epoch,
-                               node := Node} = Args0, Conf) ->
+                               node := Node} = Args0, #{writer_mod := Mod} = Conf) ->
     fun() ->
-            try osiris_writer:start(Conf) of
+            try Mod:start(Conf) of
```

Comment:
We need a behaviour module with a defined API for this as plain dynamic module calls are a pain to maintain and keep safe.